sorting/heap_sort.cpp

```diff
@@ -1,52 +1,118 @@
+/**
+ * \addtogroup sorting Sorting Algorithm
```

Comment:
```suggestion
```
since the heal sort algorithm is essentially only two files, I suggest the group should contain only these files and not the test and main functions